### PR TITLE
[MRG] Moving nilearn.decoding.objective_functions._unmask to nilearn.masking._unmask_from_to_3d_array

### PR DESCRIPTION
**Pull request by thomasbazeille**
Renaming stand-alone _unmask function for SpaceNet to _unmask_from_to_3d_array and moving it in nilearn/masking.py with the relevant tests and import changes.

The goal is to use this function later to refactor ReNA for PR#1983

### CODE REVIEW

**Comment by thomasbazeille** [comment]
@KamalakerDadi can you please review this small PR when you've time ? It's a small change necessary to refactor ReNA. Thanks!

File: nilearn/decoding/space_net.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 from sklearn.linear_model.base import _preprocess_data as center_data
 from .._utils.compat import _basestring
 from .._utils.cache_mixin import CacheMixin
-from .objective_functions import _unmask
+from nilearn.masking import _unmask_from_to_3d_array
```
Comment (KamalakerDadi) [inline nilearn/decoding/space_net.py]: Could you use relative imports please ? Like the ones above and below.

File: nilearn/decoding/space_net_solvers.py [hunk 2]
```diff
@@ -13,10 +13,11 @@
 
 from math import sqrt
 import numpy as np
+from nilearn.masking import _unmask_from_to_3d_array
```
Comment (KamalakerDadi) [inline nilearn/decoding/space_net_solvers.py]: same here about relative imports.

**Comment by KamalakerDadi** [comment]
If you can address the flake8 issues as much as you can, that would be great. I agree that there are written like that.

Otherwise, LGTM

Great thinking making them into separate PR.

File: nilearn/decoding/space_net.py (hunk 1 above)
Comment (kchawla-pi) [reply nilearn/decoding/space_net.py]: @GaelVaroquaux  and I talked about this in #1919 and  while he doesn't want existing imports to be changes, for new imports absolute imports are okay.
PEP8 standard is to use Absolute imports. The only time relative imports are considered okay if  a specific absolute import is too verbose.

Comment (KamalakerDadi) [reply nilearn/decoding/space_net.py]: +1 for relative imports again.

Either we do this or change everything to absolute imports.

**Comment by GaelVaroquaux** [comment]
> +1 for relative imports again.

> Either we do this or change everything to absolute imports.

I think that we should move to absolute imports, just to be consistent
with the style guide.

However, I am not in favor of doing a major PR to change everything at
once, because I am afraid of the huge number of conflicts that it will
create.

**Comment by thomasbazeille** [comment]
Ok so if there we can keep the new absolute imports as is, can the rest seems ok, can someone merge the PR ?

Thanks a lot for the reviews

File: nilearn/decoding/space_net.py (hunk 1 above)
Comment (GaelVaroquaux) [reply nilearn/decoding/space_net.py]: Given the recommendations of PEP8, we should gradually evolve our code to absolute imports.

Comment (KamalakerDadi) [reply nilearn/decoding/space_net.py]: Are the recommendations changed since last year ? This is what I heard when Loic was there.

**Comment by GaelVaroquaux** [comment]
LGTM. +1 for merge.

PS: next time: turn off your autopep8 :D
